### PR TITLE
Add automation scripts of APIM 2.1.0 pattern deployment in K8s

### PR DESCRIPTION
**Pull request by dewmini**
## Automation tests
 N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by kasunbg** [comment]
@pubudu538 @anuruddhal could you please check and merge this?

File: automation-scripts/common/scripts/base.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# ------------------------------------------------------------------------
+#
+# Copyright 2016 WSO2, Inc. (http://wso2.com)
```
Comment (anuruddhal) [inline automation-scripts/common/scripts/base.sh]: Fix the license header, There are few places with the wrong year.

File: automation-scripts/common/scripts/wait-until-server-starts.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+product=${PWD##*/}
+profile=$1
+port=$2
+
+echo "Waiting ${product} to launch on http://${host}"
+sleep 120
+#until $(curl --output /dev/null --silent --head --fail http://${host}:${port}); do
+#   printf '.'
+#    sleep 3
+#done
```
Comment (anuruddhal) [inline automation-scripts/common/scripts/wait-until-server-starts.sh]: Is the code correct here? AFAIU we need to remove sleep and uncomment code.

File: automation-scripts/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# limitations under the License
+
+# ------------------------------------------------------------------------
+
+export KUBERNETES_MASTER=$1
+
+# Log Message should be parsed $1
+log(){
+ TIME=`date`
+ #echo "$TIME : $1" >> "$LOG_FILE_LOCATION"
```
Comment (anuruddhal) [inline automation-scripts/deploy.sh]: Remove commented code.

File: automation-scripts/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        h)
+            showUsageAndExitDefault
+            ;;
+        \?)
+            showUsageAndExitDefault
+            ;;
+    esac
+done
+
+#kubectl create secret docker-registry registrykey --docker-server=$2 --docker-username=$3 --docker-password=$4 --docker-email=$5
```
Comment (anuruddhal) [inline automation-scripts/deploy.sh]: Shouldn't this be uncommented?

File: automation-scripts/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    ssh core@$node "docker pull docker.wso2.com/sshd-kubernetes:1.0.0 &&
+		docker pull docker.wso2.com/rsync-kubernetes:1.0.0 &&
+		docker pull docker.wso2.com/wso2am-analytics-kubernetes:2.1.0 &&
+		docker pull docker.wso2.com/wso2am-kubernetes:2.1.0 &&
+		docker pull docker.wso2.com/apim-rdbms-kubernetes:2.1.0"
+done
+
+#clone APIM kubernetes artifacts repo
+env -i git clone https://github.com/wso2/kubernetes-apim.git
+env -i git checkout tags/v2.1.0-1
```
Comment (anuruddhal) [inline automation-scripts/deploy.sh]: Shall we make the tag a parameter and default to v2.1.0-1?

**Comment by imesh** [comment]
I was thinking about these files again. If we can consider the files included in this pull request as test cases for WSO2 Kubernetes resources, may be it would make much sense to have them here. In that approach we may need to clearly create folders with proper names similar to the test folders we have in Java, Node.js, Go and other types of projects and clearly document the usage. That would allow users of these resources to run them on their environments and verify the deployment process.

**Pull request merged**